client/src/main/java/io/pravega/client/segment/impl/AsyncSegmentInputStreamImpl.java

```diff
@@ -84,9 +85,19 @@ public void connectionDropped() {
             closeConnection(new ConnectionFailedException());
         }
 
+        @SneakyThrows
```

Comment:
Specify which exception is being thrown here. Why is a new exception being added? As far as I can tell neither `updateStaleValueInCache` nor `closeConnection` throw anything.

Reply:
The call to get connection object in the below code snippet throws InterruptedException, ExecutionException.
I have added a catch and logged the message, post this we are closing the connection in finally.

ClientConnection conn = connection.get();